src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHousCsvProcessor.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Globalization;
+using CsvHelper;
+using CsvHelper.Configuration;
+using EPR.SubsidiaryBulkUpload.Application.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+
+namespace EPR.SubsidiaryBulkUpload.Application.Services;
+
+public class CompaniesHousCsvProcessor(
+    ILogger<CompaniesHousCsvProcessor> logger) : ICsvProcessor
+{
+    private readonly ILogger<CompaniesHousCsvProcessor> _logger = logger;
+
+    public async Task<int> ProcessStream(Stream stream)
+    {
+        var rowCount = 0;
+
+        using var reader = new StreamReader(stream);
+        using var csv = new CsvReader(reader, new CsvConfiguration(CultureInfo.CurrentCulture)
+        {
+            HasHeaderRecord = true,
+        });
+
+        await csv.ReadAsync();
+        csv.ReadHeader();
+        var header = csv.HeaderRecord;
+        _logger.LogInformation("Found csv header of length {Length} - {Values}", header.Length, string.Join(',', header));
+
+        while (await csv.ReadAsync())
+        {
+            rowCount++;
+
+            var field = csv.GetField(0);
+            if (rowCount % 1000 == 0)
```

Comment:
Why are we logging these this?
Wouldn't a simple log message of the number processed and the csv header do?